### PR TITLE
Split brain handling

### PR DESCRIPTION
**Pull request by ytsarev**
* Use TXT as 'heartbeat' for spefic gslb deployment
* Use UTC timestamp with second level precision
* Switch to automatically generated geo tag based fqdns
* Eventually we will detect 'stale' gslb deployments by an
  outdated timestamp
* The timestamp is about to handle situation of split brain
  between multiple gslb deployments in case of network partition
  between them but not between gslb and edgeDNS

### CODE REVIEW

**Comment by ytsarev** [comment]
@donovanmuller it's not completed implementation. Before proceeding further I would like to validate the design.

**Comment by donovanmuller** [comment]
@ytsarev 👍let's discuss and walk through the permutations.

File: pkg/controller/gslb/gslb_controller_test.go [hunk 1]
```diff
@@ -310,6 +327,87 @@ func TestGslbController(t *testing.T) {
...
+	})
+
+	t.Run("Can check external Gslb TXT record for validity", func(t *testing.T) {
+		extClusters := getExternalClusterFQDNs(gslb)
+
+		delegateTo := []ibclient.NameServer{
+			{Address: "10.0.0.1", Name: "test-gslb-ns-eu.example.com"},
+			{Address: "10.0.0.2", Name: "test-gslb-ns-eu.example.com"},
+			{Address: "10.0.0.3", Name: "test-gslb-ns-eu.example.com"},
+			{Address: "10.1.0.1", Name: "test-gslb-ns-sa.example.com"},
```
Comment (donovanmuller) [inline pkg/controller/gslb/gslb_controller_test.go]: County code for South Africa is `za` 😀

**Comment by ytsarev** [comment]
@donovanmuller commits are squashed to more or less important semantic sets.

Btw that's how relevant part of test suite execution looks like
```
=== RUN   TestGslbController/Can_check_external_Gslb_TXT_record_for_validity_and_fail_if_it_is_expired
2020-02-27T14:43:55.314+0100	INFO	controller_gslb	Query for TXT test-gslb-ns-eu.example.com.

2020-02-27T14:43:55.314+0100	INFO	controller_gslb	TXTs found: [2020-02-27T13:33:55]

2020-02-27T14:43:55.314+0100	INFO	controller_gslb	Split brain TXT raw record: test-gslb-ns-eu.example.com.	3600	IN	TXT	"2020-02-27T13:33:55"
2020-02-27T14:43:55.314+0100	INFO	controller_gslb	Split brain TXT raw time stamp: 2020-02-27T13:33:55
2020-02-27T14:43:55.314+0100	INFO	controller_gslb	Split brain TXT parsed time stamp: 2020-02-27 13:33:55 +0000 UTC
2020-02-27T14:43:55.314+0100	INFO	controller_gslb	Split brain TXT time diff: 10m0.314661s
=== RUN   TestGslbController/Can_check_external_Gslb_TXT_record_for_validity_and_pass_if_it_is_not_expired
2020-02-27T14:43:55.314+0100	INFO	controller_gslb	Query for TXT test-gslb-ns-za.example.com.

2020-02-27T14:43:55.314+0100	INFO	controller_gslb	TXTs found: [2020-02-27T13:40:55]

2020-02-27T14:43:55.314+0100	INFO	controller_gslb	Split brain TXT raw record: test-gslb-ns-za.example.com.	3600	IN	TXT	"2020-02-27T13:40:55"
2020-02-27T14:43:55.314+0100	INFO	controller_gslb	Split brain TXT raw time stamp: 2020-02-27T13:40:55
2020-02-27T14:43:55.314+0100	INFO	controller_gslb	Split brain TXT parsed time stamp: 2020-02-27 13:40:55 +0000 UTC
2020-02-27T14:43:55.315+0100	INFO	controller_gslb	Split brain TXT time diff: 3m0.315007s
```

**Pull request merged**